source/game/p_client.cpp

```diff
@@ -798,6 +799,24 @@ void ClientBegin( edict_t *ent )
 	G_Gametype_ScoreEvent( client, "enterGame", NULL );
 }
 
+void ClientAuth ( edict_t *ent, uint64_t steamid )
+{
+	gclient_t *client = ent->r.client;
+	client->steamid = steamid;
+	client->authenticated = true;
+
+
+	char buf[17];
+	Q_snprintfz(buf, sizeof(buf), "%llu", steamid);
+
+	if ( Q_strrstr(g_permanent_operators->string, buf)){
```

Comment:
given the `g_permanent_operators` setting: `123456,65432`
If someone has the steam id `123` he will be operator with this code, but he is not in this list.

Reply:
right, forgot about that case.
"%017llu" instead would be safe here right?

Reply:
That means all operators in `g_permanent_operators` need to be zero-padded as well.

Reply:
Instead, just tokenize the string, parse each number and straight out compare it against the steam id as number.

Reply:
https://github.com/xPaw/SteamID.php - Some examples of converting between Steam ID types. Pretty certain that SteamID64 is ALWAYS 17 digits. Might be easiest to only support SteamID64 and ignore any comma-separated values of a different length.

Reply:
> https://github.com/xPaw/SteamID.php - Some examples of converting between Steam ID types. Pretty certain that SteamID64 is ALWAYS 17 digits. Might be easiest to only support SteamID64 and ignore any values of a different length.

https://developer.valvesoftware.com/wiki/SteamID figured I would share this incase you were curious. It covers everything related to steamids.

Reply:
should be fixed